handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -836,98 +864,74 @@ public final SSLEngineResult unwrap(
                         srcsOffset++;
                         continue;
                     }
+
                     // Write more encrypted data into the BIO. Ensure we only read one packet at a time as
                     // stated in the SSLEngine javadocs.
-                    int written = writeEncryptedData(src, Math.min(packetLengthRemaining, src.remaining()));
+                    int written = writeEncryptedData(src, Math.min(packetLength, remaining));
                     if (written > 0) {
-                        packetLengthRemaining -= written;
-                        if (packetLengthRemaining == 0) {
-                            // A whole packet has been consumed.
-                            break;
-                        }
+                        packetLength -= written;
+                        bytesConsumed += written;
```

Comment:
I think the current code is a bit cleaner and easier to understand. I can check if it makes any difference tho if you feel strong about it